From e2cb6e8db69e96c1514c2992e2d4fd6c8c1b8820 Mon Sep 17 00:00:00 2001
From: Bean Huo <beanhuo@micron.com>
Date: Sun, 8 Jan 2023 23:40:57 +0100
Subject: scsi: ufs: core: bsg: Fix cast to restricted __be16 warning

Fix the following sparse endianness warning:

"sparse warnings: drivers/ufs/core/ufs_bsg.c:91:25: sparse: sparse: cast to
restricted __be16."

For consistency with endianness annotations of other UFS data structures,
change __u16/32 to __be16/32 in UFS ARPMB data structures.

Fixes: 6ff265fc5ef6 ("scsi: ufs: core: bsg: Add advanced RPMB support in ufs_bsg")
Reported-by: kernel test robot <lkp@intel.com>
Signed-off-by: Bean Huo <beanhuo@micron.com>
Reviewed-by: Bart Van Assche <bvanassche@acm.org>
Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
---
 include/uapi/scsi/scsi_bsg_ufs.h | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/include/uapi/scsi/scsi_bsg_ufs.h b/include/uapi/scsi/scsi_bsg_ufs.h
index 276e2772328fc..2801b65299aa3 100644
--- a/include/uapi/scsi/scsi_bsg_ufs.h
+++ b/include/uapi/scsi/scsi_bsg_ufs.h
@@ -97,18 +97,18 @@ struct utp_upiu_req {
 };
 
 struct ufs_arpmb_meta {
-	__u16	req_resp_type;
+	__be16	req_resp_type;
 	__u8	nonce[16];
-	__u32	write_counter;
-	__u16	addr_lun;
-	__u16	block_count;
-	__u16	result;
+	__be32	write_counter;
+	__be16	addr_lun;
+	__be16	block_count;
+	__be16	result;
 } __attribute__((__packed__));
 
 struct ufs_ehs {
 	__u8	length;
 	__u8	ehs_type;
-	__u16	ehssub_type;
+	__be16	ehssub_type;
 	struct ufs_arpmb_meta meta;
 	__u8	mac_key[32];
 } __attribute__((__packed__));
-- 
cgit 

